src/Adapter/LegacyHookSubscriber.php

```diff
@@ -278,18 +277,17 @@ public function __call($name, $args)
         $ids = explode('_', $name);
         array_shift($ids); // remove 'call'
 
-        if (count($ids) != 2) {
+        if (count($ids) !== 2) {
             throw new \BadMethodCallException('The call to \''.$name.'\' is not recognized.');
         }
 
-        $moduleId = $ids[1];
+        $moduleId = (int) $ids[1];
+        list($event, $hookName) = $args;
 
-        $hookName = $args[1];
-        $event = $args[0];
         /* @var $event HookEvent */
         $content = Hook::exec($hookName, $event->getHookParameters(), $moduleId, ($event instanceof RenderingHookEvent));
 
-        if ($event instanceof RenderingHookEvent) {
+        if ($event instanceof RenderingHookEvent && 0 !== $moduleId) {
```

Comment:
haha! So you brought the same change as @kpodemski in https://github.com/PrestaShop/PrestaShop/pull/8463

You wanted to log that case too, remember?

Reply:
it's not the same issue, I have added "0" as a specific use case to allow the Hook Collector to collect hooks that are not collected if no modules use it.

#8463 solves the "undefined module id" when mine completes the "module with id 0" I've introduced to allow the DataCollector to get all hooks.